### PR TITLE
Separate onboarding concerns into separate classes

### LINKED ISSUE #8266: Clean up src/Features/Onboarding.php
- clean up `src/Features/Onboarding.php`
- 1000+ lines without clear responsibility
- note that this is currently being used so cleanup needs to be done carefully
- use the "move method" refactor where necessary and make new classes internal

My estimate: 5 hours

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #8266 

This PR breaks the onboarding class into manageable logic and moves it into an internal namespace.

Note that while a lot of the logic could be cleaned up here, this PR's primary focus is separation as opposed to a full refactor in logic.  Potential follow-ups:

* Excessive preloaded data
* Onboarding profile data API
* Onboarding steps API
* Tests
* Static initialization may benefit from DI

### Detailed test instructions:

1. Smoke test all aspects of the onboarding setup wizard and task list (including hiding/showing a task list by making use of the help menu)
2. Check that the separation makes sense.  One alternative is a breakdown of concerns by individual steps.
3. Walk through the changes and make sure they make sense.  Note that the first commit is mostly file separation without much logic change - https://github.com/woocommerce/woocommerce-admin/pull/8280/commits/5854096669ca341edfd8b1bd968d61ec2e9f8229 ; Changes in logic were committed separately to allow easier reviewing.
4. Attempt to trigger various notes that use onboarding information
5. Carefully check the debug log for any errors that might occur

### CODE REVIEW

**Comment by joshuatf** [comment]
Just a quick note on this one on some items needed prior to merge:

* It will need a deprecation facade around some of the `Onboarding` methods.  `should_show_tasks()` was found in Google Listings and Ads.  There could be others.
* Instances should be created from classes and private methods used where possible.  This may be okay for a follow-up given a lot of this existed already, but might be good to go ahead and consider here.
* Folder should be changed to `src-internal/Admin` to match new direction.

**Comment by louwie17** [comment]
> Just a quick note on this one on some items needed prior to merge:
> 
> * It will need a deprecation facade around some of the `Onboarding` methods.  `should_show_tasks()` was found in Google Listings and Ads.  There could be others.
> * Instances should be created from classes and private methods used where possible.  This may be okay for a follow-up given a lot of this existed already, but might be good to go ahead and consider here.
> * Folder should be changed to `src-internal/Admin` to match new direction.

I have updated this PR and addressed all three of these items.
Just a note that I actually did remove the `should_show_tasks()` function as it was already set to be depreciated and that is how Google Listings was handling it as well (only running it if WC was below version 5.9). Having set that, I did leave a depreciated `Onboarding` class with several depreciated functions that point to the new one's.

**Review by jacob-sewell — COMMENTED** [review]
I'm not seeing any errors, but I'm also not seeing the task list after going through the initial setup. I ran out of time today to look for other factors tonight.

**Comment by becdetat** [comment]
> I'm not seeing any errors, but I'm also not seeing the task list after going through the initial setup

@jacob-sewell I could be totally off-base but a new feature flag has been added to the configuration so you might need to run the `build` task (`pnpm build` I think?) to get the flag pushed through.

File: src-internal/Admin/Onboarding/Onboarding.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace Automattic\WooCommerce\Internal\Admin\Onboarding;
+
+/**
+ * Initializes backend logic for the onboarding process.
+ */
+class Onboarding {
+	/**
+	 * Initialize onboarding functionality.
+	 */
+	public static function init() {
```
Comment (becdetat) [inline src-internal/Admin/Onboarding/Onboarding.php:16]: Love this

**Review by octaedro — APPROVED** [review]
Great job @louwie17 finishing this up! It's a pretty massive PR. This is testing well here and code LGTM 🚀

**Pull request merged**